### ISSUE TITLE
[BUG] 'Invalid schema' error, with a yaml schema converted from v2 to v3 

### ISSUE THREAD

**Issue by aSemy** [schemathesis/schemathesis#1140]
**Describe the bug**

I'd like to do stateful testing against the Petstore v2 API. For this, I'd like to create the `response.links:` elements. I used Swagger's editor to upgrade the schema from v2 to v3, but I am unable to load the schema

**To Reproduce**

1. Find the yaml schema of petstore v2 https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/2_0/petstore.yaml
1. Go to https://editor.swagger.io/
1. File > Import from URL > paste in a link to the raw v2 schema https://raw.githubusercontent.com/OpenAPITools/openapi-generator/master/modules/openapi-generator/src/test/resources/2_0/petstore.yaml
2. Edit > Convert to OpenAPI 3 > Convert
3. Copy the output (`openapi: 3.0.1`)
4. Go to https://validator.swagger.io/, paste the schema into the `POST /` endpoint (with request body as application/yml), verify it is valid
5. Save the converted schema to a local file, `./schemas/petstore_v3.yaml`
6. Try loading the schema 
```python
import schemathesis

schema = schemathesis.from_path(
    path="./schemas/petstore_v3.yaml",
    base_url="http://localhost:8080/v2",
)
```
7. Error message: `jsonschema.exceptions.ValidationError: Invalid schema`
Cause: 
```
  File ".../python@3.9/3.9.4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/re.py", line 201, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or bytes-like object
```

**Expected behavior**

I would like to see a clearer error message about why this schema is invalid.

I would also like to be able to load schemas that validator.swagger.io reports as valid.

**Environment (please complete the following information):**

 - OS: macOS 10.15.7
 - Python version: 3.9.4
 - schemathesis 3.6.3, schemathesis-test 0.1.0
 - Spec version: Open API 3.0.1

**Additional context**

As a workaround, I can use [yq](https://mikefarah.gitbook.io/yq/) to convert yaml to json. 

```console
cat petstore_converted.yaml | yq > petstore_converted.json
```

```python
import schemathesis

schema = schemathesis.from_path(
    # path="./schemas/petstore_converted.yaml",
    path="./schemas/petstore_converted.json",
    base_url="http://localhost:8080/v2",
)
print(schema.raw_schema)
```

Output: 
```console
{'openapi': '3.0.1', 'info': {'title': 'Swagger Petstore', ...
Process finished with exit code 0
```

**Comment by Stranger6667** [comment]
Hi @aSemy 

Thank you for the detailed report! 

At the moment, Schemathesis relies on the `jsonschema` behavior on that matter, and this problem is caused by YAML parsing behavior and the `jsonschema` approach to such input. It was discussed in `jsonschema` [here](https://github.com/Julian/jsonschema/pull/286).

In short - there were multiple PRs to improve these, but they were closed. The intend is that the input should be deserialized valid JSON, and any transformations should be done before passing it to `validate`; however, `pyyaml` implicitly allows it :/ 

It is definitely not a pleasant user experience, and I plan to solve it in the future. There is [a tracking issue](https://github.com/schemathesis/schemathesis/issues/834) for it - it takes much longer than I anticipated, though :(

**Comment by Stranger6667** [comment]
As for the validation behavior on https://validator.swagger.io/.

The OpenAPI spec [explicitly requires](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#patterned-fields-1) status codes to be strings:

> ... This field MUST be enclosed in quotation marks (for example, "200") for compatibility between JSON and YAML...

**Comment by aSemy** [comment]
Thanks for looking into it! And for the good explanation. 

I'm happy to close this issue, as there's another issue for tracking improved error messages.
